### PR TITLE
Add build process with gulp to generate .pot

### PR DESCRIPTION
**Pull request by bahiirwa**
- [x] Adds build process with gulp to generate .pot
- [x] Generates current string up for translation.

Run `npm i && gulp watch` to generate newly translated files.
This would be a stepping block to help the move the project from English base only.
There a number of string to make translatable as well in the project. This can be another PR to make it manageable to evaluate.

### CODE REVIEW

**Comment by smileBeda** [comment]
The strings are all localised (using core localisation functions)

But as I understand your PR is to... add actual translations?
I am not so familiar with these things, sorry the dumb question.

I changed a bunch of files and such in v1.3 (not yet released), as well changed some cases to traits.
Do such changes affect what you PR'd? I can see in the PR there are some references to filenames. That is what the script generates for you?
There are many more strings in the plugin, all localised with some form of `_()` function, but I do not see them there.

I probably do not understand the tool enough. I will play with it after installing NPM (sorry, I am not that much around the terminal 🦈 )

Thanks!

File: readme.md [hunk 1]
```diff
@@ -33,6 +33,9 @@ The conditions for this to work are:
 
 ## Changelog
 
+### 1.2.1
+[Fixed] Build Process to generate .pot files.
```
Comment (smileBeda) [inline readme.md]: Can we change this to `[Added]`

File: changelog.txt [hunk 2]
```diff
@@ -1,3 +1,6 @@
+= 1.2.1 =
+[Fixed] Build Process to generate .pot files.
```
Comment (smileBeda) [inline changelog.txt]: Can we change this to `[Added]`

**Comment by bahiirwa** [comment]
> The strings are all localised (using core localisation functions)

Yes, that is what should be even with this PR merged. This is a build process step to ally the .pot file to be generated so other languages can copy and transalte to their locale. We need to continue using the WP localization functions.

> But as I understand your PR is to... add actual translations?

No, it is not. It is to add a process to make the .pot file generated.

> I changed a bunch of files and such in v1.3 (not yet released), as well changed some cases to traits. Do such changes affect what you PR'd? I can see in the PR there are some references to filenames. That is what the script generates for you?

I have generated the file with 1.3.0 files. I pulled main branch. Yes, the script does this for you.

> There are many more strings in the plugin, all localised with some form of `_()` function, but I do not see them there.

Let me check again. I wish I had an example to check this.

Please try to install node and npm. I found this easier to use than using the .pot softwares that have a price tag sometimes.

**Comment by bahiirwa** [comment]
I have also localized some more strings to test the process.
All the files/directory allow for changes to be made to .pot as you run `gulp watch` in terminal

Let me know how your tests go. We can them work on localizing the other strings.

**Comment by smileBeda** [comment]
I am not able to test this.

1. I installed NPM (had it already actually so all good here)
2. Downloaded the https://github.com/bahiirwa/tukutoi-cp-directory-integration/tree/fix-i18n
3. `cd`'d into that folder
4. Run `npm i && gulp watch`

This threw errors:
```
npm WARN saveError ENOENT: no such file or directory, open '.../wp-content/plugins/tukutoi-cp-directory-integration/package.json'
npm WARN enoent ENOENT: no such file or directory, open '.../wp-content/plugins/tukutoi-cp-directory-integration/package.json'
npm WARN tukutoi-cp-directory-integration No description
npm WARN tukutoi-cp-directory-integration No repository field.
npm WARN tukutoi-cp-directory-integration No README data
npm WARN tukutoi-cp-directory-integration No license field.

added 18 packages from 13 contributors in 1.108s
found 0 vulnerabilities

[15:33:47] Local gulp not found in .../wp-content/plugins/tukutoi-cp-directory-integration
[15:33:47] Try running: npm install gulp
(base) ... tukutoi-cp-directory-integration % 
```

So (`gulp` is in my `/usr/local/bin/gulp` - thus upstream from my working folder... but anyway I...) ...run `npm install gulp` _inside_ this directory as it seems to be needed _inside_ and not upstream.

This however also threw errors:
```
npm WARN deprecated chokidar@2.1.8: Chokidar 2 will break on node v14+. Upgrade to chokidar 3 with 15x less dependencies.
npm WARN deprecated source-map-resolve@0.5.3: See https://github.com/lydell/source-map-resolve#deprecated
npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
npm WARN deprecated source-map-url@0.4.1: See https://github.com/lydell/source-map-url#deprecated
npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
npm WARN deprecated fsevents@1.2.13: fsevents 1 will break on node v14+ and could be using insecure binaries. Upgrade to fsevents 2.

> fsevents@1.2.13 install .../wp-content/plugins/tukutoi-cp-directory-integration/node_modules/fsevents
> node install.js

No receipt for 'com.apple.pkg.CLTools_Executables' found at '/'.

No receipt for 'com.apple.pkg.DeveloperToolsCLILeo' found at '/'.

No receipt for 'com.apple.pkg.DeveloperToolsCLI' found at '/'.

gyp: No Xcode or CLT version detected!
gyp ERR! configure error 
gyp ERR! stack Error: `gyp` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onCpExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/configure.js:351:16)
gyp ERR! stack     at ChildProcess.emit (events.js:315:20)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:275:12)
gyp ERR! System Darwin 21.5.0
gyp ERR! command "/usr/local/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd .../wp-content/plugins/tukutoi-cp-directory-integration/node_modules/fsevents
gyp ERR! node -v v12.18.3
gyp ERR! node-gyp -v v5.1.0
gyp ERR! not ok 
npm WARN saveError ENOENT: no such file or directory, open '.../wp-content/plugins/tukutoi-cp-directory-integration/package.json'
npm WARN enoent ENOENT: no such file or directory, open '.../wp-content/plugins/tukutoi-cp-directory-integration/package.json'
npm WARN tukutoi-cp-directory-integration No description
npm WARN tukutoi-cp-directory-integration No repository field.
npm WARN tukutoi-cp-directory-integration No README data
npm WARN tukutoi-cp-directory-integration No license field.

+ gulp@4.0.2
added 311 packages from 219 contributors and audited 343 packages in 17.243s

10 packages are looking for funding
  run `npm fund` for details

found 1 high severity vulnerability
  run `npm audit fix` to fix them, or `npm audit` for details
(base) ... tukutoi-cp-directory-integration % 
```
`npm audit` threw this:
```
npm ERR! code EAUDITNOPJSON
npm ERR! audit No package.json found: Cannot audit a project without a package.json

npm ERR! A complete log of this run can be found in:
npm ERR!    .../.npm/_logs/2022-07-03T08_38_07_800Z-debug.log
```

Running `npm i && gulp watch` again at this point (expectedly) continues to throw errors:
```
npm WARN saveError ENOENT: no such file or directory, open '.../wp-content/plugins/tukutoi-cp-directory-integration/package.json'
npm WARN enoent ENOENT: no such file or directory, open '.../wp-content/plugins/tukutoi-cp-directory-integration/package.json'
npm WARN tukutoi-cp-directory-integration No description
npm WARN tukutoi-cp-directory-integration No repository field.
npm WARN tukutoi-cp-directory-integration No README data
npm WARN tukutoi-cp-directory-integration No license field.

audited 548 packages in 1.588s

10 packages are looking for funding
  run `npm fund` for details

found 4 high severity vulnerabilities
  run `npm audit fix` to fix them, or `npm audit` for details
[15:40:11] No gulpfile found
```

It seems there's some file missing?

**Comment by smileBeda** [comment]
> I have generated the file with 1.3.0 files. I pulled main branch. Yes, the script does this for you.

I don't think this is the case.
I see many missing files in the [fix-i18n](https://github.com/bahiirwa/tukutoi-cp-directory-integration/tree/fix-i18n) branch (however that should probably not be an issue, just pointing it out because the fix-i18n seems based on _master_ not 1.3.0 which is not released or merged yet (see https://github.com/TukuToi/tukutoi-cp-directory-integration/tree/1.3.0)
That is work in progress and it adds several new files/strings.

---
I think I got the gulp watch working, after adding the missing files (I took them from your commit, as somehow they where missing in the https://github.com/bahiirwa/tukutoi-cp-directory-integration/tree/fix-i18n branch after I checked that out)
So it seems to work now, I added a couple new strings and that generated some new .pot entries, so that seems to work.

From there, how do I "stop" gulp from watching?
I basically just close the terminal and that's it?

**Comment by smileBeda** [comment]
Weird, when I look at your online branch I see the gulp and other files.
When I clone your branch, locally I do not have those files.

File: package.json [hunk 3]
```diff
@@ -0,0 +1,16 @@
+{
+	"name": "tukutoi-cp-directory-integration",
+	"version": "0.1.0",
```
Comment (smileBeda) [inline package.json:3]: Pleas apologise my ignorance - should this not state the current Plugin version, so translations are updated if the version changes? I saw the same in the -lock file

Thanks!

**Comment by bahiirwa** [comment]
Command C or Ctrl C stops the gulp running. If you want to have a one time run, use gulp wp_pot.

That's interesting that files are missing. No need to get my commits if they are funny. You can implement if this desired via your branch

File: languages/cp-plgn-drctry.pot [hunk 4]
```diff
@@ -0,0 +1,75 @@
+# Copyright (C) 2022 Cp_Plgn_Drctry
```
Comment (smileBeda) [inline languages/cp-plgn-drctry.pot:1]: Given that the up next version is 1.3.0 I think we can exclude this entire content. 
We can generate the .po just before releasing 1.3 (there are many string changes in there)

Of course we can also include it now and then just regenerate it later.

File: package.json (hunk 3 above)
Comment (bahiirwa) [reply package.json:3]: Feel free to update that. But it doesn't matter for the localization and generating of the .pot file. That's versioning of the build process.

File: languages/cp-plgn-drctry.pot (hunk 4 above)
Comment (bahiirwa) [reply languages/cp-plgn-drctry.pot:1]: Of course. The important thing is if the build makes the work easy. We can always regenerate the file when ready.

**Comment by smileBeda** [comment]
Simone showed me that with `wp-cli` you can do the POT file with a single command (and there's no need for the gulp/npm)

Being wp-cli a more "WP way" to do things, and additionally you can run it from anywhere without additional libraries or files added to the plugin, I am going with that.

Closing this PR - but thank you a lot for the heads up that a POT file actually only makes sense if it has some content that a translator can translate!!

**Pull request closed without merging**